core/base/src/TROOT.cxx

```diff
@@ -1250,7 +1250,7 @@ void TROOT::EndOfProcessCleanups()
    CloseFiles();
 
    if (gInterpreter) {
-      gInterpreter->ShutDown();
+      gInterpreter->ResetGlobals();
```

Comment:
Why we have this call here. The ShutDown call should be a superset of ResetGlobals. IIRC, the commit message of this explained it.

Reply:
We want the interpreter global (which have stack like semantic) to be deleted **before** the next set of deletions (line 1264-1269) **BUT** those same deletions **must** have a functioning interpreter.